### PR TITLE
mrc-3126 HINT to send project state to hintr with generate output zip request

### PR DESCRIPTION
**Pull request by LekanAnni**
## Description

This PR adds project state payload to download spectrum endpoint.

## Type of version change
_Delete as appropriate_

Minor 

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

**Review by EmmaLRussell — CHANGES_REQUESTED** [review]
This seems like a good approach to me in general. 

I generated the output zip for a completed project and saw the below `project_state.json`. It looks like there's some missing stuff that isn't getting included correctly: the calibrate options aren't included for some reason and neither are the download ids.

However it occurred me that of course we don't have the download ids when we make this request, because this *is* the request for the download id! And we don't have any of them because they all get requested at once when the page loads. In fact if we were to include any download ids here, they would be wrong because they would be out of date (though I think they get cleared on recalibrate). Thoughts on this, @r-ash ? Do we really need to include the download ids? We could just let the rehydrated project generate its own couldn't we?
```
{
    "datasets": {
        "pjnz": {
            "path": "uploads/72A9B1F58AAA743ADA64C6AE985CF228.pjnz",
            "filename": "demo_mwi2019.pjnz"
        },
        "population": {
            "path": "uploads/651105353D7153381ED29363DF0D772F.csv",
            "filename": "demo_population_agesex.csv"
        },
        "shape": {
            "path": "uploads/EBE533976BFAF0CABCA2C2E1B611B9C7.geojson",
            "filename": "demo_areas.geojson"
        },
        "survey": {
            "path": "uploads/F669CA9AA38A3993B5A9E9D3EB717C7D.csv",
            "filename": "demo_survey_hiv_indicators.csv"
        },
        "programme": {
            "path": "uploads/8301300AB39BE177FA593571B9DD94C4.csv",
            "filename": "demo_art_number.csv"
        },
        "anc": {
            "path": "uploads/E6323AAEB045D31E4A267398F669CF20.csv",
            "filename": "demo_anc_testing.csv"
        }
    },
    "model_fit": {
        "options": {
            "area_scope": "MWI",
            "area_level": "4",
            "calendar_quarter_t2": "CY2021Q4",
            "calendar_quarter_t3": "CY2022Q3",
            "calendar_quarter_t1": "CY2016Q1",
            "survey_prevalence": "DEMO2016PHIA",
            "survey_art_coverage": "DEMO2016PHIA",
            "anc_clients_year2": "",
            "anc_clients_year2_num_months": "12",
            "anc_prevalence_year1": "2016",
            "anc_prevalence_year2": "",
            "anc_art_coverage_year1": "2016",
            "anc_art_coverage_year2": "",
            "include_art_t1": "true",
            "include_art_t2": "true",
            "artattend": "true",
            "artattend_t2": "true",
            "spectrum_population_calibration": "subnational",
            "output_aware_plhiv": "true",
            "max_iterations": 250,
            "no_of_samples": 1000,
            "rng_seed": 28,
            "artattend_log_gamma_offset": -4,
            "rho_paed_x_term": "false",
            "rho_paed_15to49f_ratio": "true",
            "alpha_xst_term": "false",
            "adjust_area_growth": "false",
            "use_survey_aggregate": "false",
            "survey_recently_infected": [

            ],
            "use_kish_prev": "true",
            "deff_prev": 1,
            "use_kish_artcov": "true",
            "deff_artcov": 1,
            "use_kish_recent": "true",
            "deff_recent": 1
        },
        "id": "deccb7d8575f28d4991b754341cb874d"
    },
    "calibrate": {
        "options": {

        },
        "id": "a16a664678ad9e8dcd567abca2401d1a"
    },
    "model_output": {
        "id": ""
    },
    "coarse_output": {
        "id": ""
    },
    "summary_report": {
        "id": ""
    },
    "comparison_report": {
        "id": ""
    },
    "version": {
        "hintr": "1.0.35",
        "naomi": "2.6.26",
        "rrq": "0.5.6",
        "traduire": "0.0.6"
    }
}
```

**Comment by r-ash** [comment]
> we don't have the download ids when we make this request

:facepalm: yeah of course not! I can update hintr side to remove those

**Comment by r-ash** [comment]
> :facepalm:  yeah of course not! I can update hintr side to remove those

See https://github.com/mrc-ide/hintr/pull/386

**Comment by LekanAnni** [comment]
{
    "datasets": {
        "pjnz": {
            "path": "uploads/72A9B1F58AAA743ADA64C6AE985CF228.pjnz",
            "filename": "demo_mwi2019.pjnz"
        },
        "population": {
            "path": "uploads/651105353D7153381ED29363DF0D772F.csv",
            "filename": "demo_population_agesex.csv"
        },
        "shape": {
            "path": "uploads/EBE533976BFAF0CABCA2C2E1B611B9C7.geojson",
            "filename": "demo_areas.geojson"
        },
        "survey": {
            "path": "uploads/F669CA9AA38A3993B5A9E9D3EB717C7D.csv",
            "filename": "demo_survey_hiv_indicators.csv"
        },
        "programme": {
            "path": "uploads/8301300AB39BE177FA593571B9DD94C4.csv",
            "filename": "demo_art_number.csv"
        },
        "anc": {
            "path": "uploads/E6323AAEB045D31E4A267398F669CF20.csv",
            "filename": "demo_anc_testing.csv"
        }
    },
    "model_fit": {
        "options": {
            "area_scope": "MWI",
            "area_level": "4",
            "calendar_quarter_t2": "CY2021Q4",
            "calendar_quarter_t3": "CY2022Q3",
            "calendar_quarter_t1": "CY2016Q1",
            "survey_prevalence": "DEMO2016PHIA",
            "survey_art_coverage": "DEMO2016PHIA",
            "anc_clients_year2": "",
            "anc_clients_year2_num_months": "12",
            "anc_prevalence_year1": "2016",
            "anc_prevalence_year2": "",
            "anc_art_coverage_year1": "2016",
            "anc_art_coverage_year2": "",
            "include_art_t1": "true",
            "include_art_t2": "true",
            "artattend": "true",
            "artattend_t2": "true",
            "spectrum_population_calibration": "subnational",
            "output_aware_plhiv": "true",
            "max_iterations": 250,
            "no_of_samples": 1000,
            "rng_seed": 28,
            "artattend_log_gamma_offset": -4,
            "rho_paed_x_term": "false",
            "rho_paed_15to49f_ratio": "true",
            "alpha_xst_term": "false",
            "adjust_area_growth": "false",
            "use_survey_aggregate": "false",
            "survey_recently_infected": [

            ],
            "use_kish_prev": "true",
            "deff_prev": 1,
            "use_kish_artcov": "true",
            "deff_artcov": 1,
            "use_kish_recent": "true",
            "deff_recent": 1
        },
        "id": "a40bf0547c5b3f41bf734edd3cf8ef5b"
    },
    "calibrate": {
        "options": {
            "controlSections": [
                {
                    "label": "Calibration options",
                    "description": "Options for calibrating to Spectrum outputs. If national Spectrum file, 'National' and 'Subnational' options will return the same result. Some calibrations will invalidate model inference to survey, ART programme, and ANC testing data. If calibration adjustments are large, it is recommended to review Spectrum results and Naomi input data rather than accept large calibration adjustments.",
                    "controlGroups": [
                        {
                            "label": "Adjust to Spectrum PLHIV",
                            "controls": [
                                {
                                    "name": "spectrum_plhiv_calibration_level",
                                    "label": "Level",
                                    "type": "select",
                                    "required": true,
                                    "options": [
                                        {
                                            "id": "none",
                                            "label": "None"
                                        },
                                        {
                                            "id": "national",
                                            "label": "National"
                                        },
                                        {
                                            "id": "subnational",
                                            "label": "Subnational"
                                        }
                                    ],
                                    "value": "national",
                                    "helpText": "Calibrate to Spectrum mean total PLHIV."
                                },
                                {
                                    "name": "spectrum_plhiv_calibration_strat",
                                    "label": "Stratification",
                                    "type": "select",
                                    "required": true,
                                    "options": [
                                        {
                                            "id": "age_coarse",
                                            "label": "Age <15 / 15+"
                                        },
                                        {
                                            "id": "sex_age_coarse",
                                            "label": "Sex and age <15 / 15+"
                                        },
                                        {
                                            "id": "age_group",
                                            "label": "5-year age group"
                                        },
                                        {
                                            "id": "sex_age_group",
                                            "label": "Sex and 5-year age group"
                                        }
                                    ],
                                    "value": "sex_age_coarse",
                                    "helpText": "Calibrate to Spectrum mean total PLHIV."
                                }
                            ]
                        },
                        {
                            "label": "Adjust to Spectrum number on ART",
                            "controls": [
                                {
                                    "name": "spectrum_artnum_calibration_level",
                                    "label": "Level",
                                    "type": "select",
                                    "required": true,
                                    "options": [
                                        {
                                            "id": "none",
                                            "label": "None"
                                        },
                                        {
                                            "id": "national",
                                            "label": "National"
                                        },
                                        {
                                            "id": "subnational",
                                            "label": "Subnational"
                                        }
                                    ],
                                    "value": "national",
                                    "helpText": "Calibrate to Spectrum mean total on ART"
                                },
                                {
                                    "name": "spectrum_artnum_calibration_strat",
                                    "label": "Stratification",
                                    "type": "select",
                                    "required": true,
                                    "options": [
                                        {
                                            "id": "age_coarse",
                                            "label": "Age <15 / 15+"
                                        },
                                        {
                                            "id": "sex_age_coarse",
                                            "label": "Sex and age <15 / 15+"
                                        },
                                        {
                                            "id": "age_group",
                                            "label": "5-year age group"
                                        },
                                        {
                                            "id": "sex_age_group",
                                            "label": "Sex and 5-year age group"
                                        }
                                    ],
                                    "value": "sex_age_coarse",
                                    "helpText": "Calibrate to Spectrum mean total on ART"
                                }
                            ]
                        },
                        {
                            "label": "Adjust to Spectrum number aware of HIV status",
                            "controls": [
                                {
                                    "name": "spectrum_aware_calibration_level",
                                    "label": "Level",
                                    "type": "select",
                                    "required": true,
                                    "options": [
                                        {
                                            "id": "none",
                                            "label": "None"
                                        },
                                        {
                                            "id": "national",
                                            "label": "National"
                                        },
                                        {
                                            "id": "subnational",
                                            "label": "Subnational"
                                        }
                                    ],
                                    "value": "national",
                                    "helpText": "Calibrate to Spectrum mean total number aware of HIV status from Shiny90"
                                },
                                {
                                    "name": "spectrum_aware_calibration_strat",
                                    "label": "Stratification",
                                    "type": "select",
                                    "required": true,
                                    "options": [
                                        {
                                            "id": "age_coarse",
                                            "label": "Age <15 / 15+"
                                        },
                                        {
                                            "id": "sex_age_coarse",
                                            "label": "Sex and age <15 / 15+"
                                        },
                                        {
                                            "id": "age_group",
                                            "label": "5-year age group"
                                        },
                                        {
                                            "id": "sex_age_group",
                                            "label": "Sex and 5-year age group"
                                        }
                                    ],
                                    "value": "sex_age_coarse",
                                    "helpText": "Calibrate to Spectrum mean total number aware of HIV status from Shiny90"
                                }
                            ]
                        },
                        {
                            "label": "Adjust to Spectrum number of new infections",
                            "controls": [
                                {
                                    "name": "spectrum_infections_calibration_level",
                                    "label": "Level",
                                    "type": "select",
                                    "required": true,
                                    "options": [
                                        {
                                            "id": "none",
                                            "label": "None"
                                        },
                                        {
                                            "id": "national",
                                            "label": "National"
                                        },
                                        {
                                            "id": "subnational",
                                            "label": "Subnational"
                                        }
                                    ],
                                    "value": "national",
                                    "helpText": "Calibrate to Spectrum mean total new infections"
                                },
                                {
                                    "name": "spectrum_infections_calibration_strat",
                                    "label": "Stratification",
                                    "type": "select",
                                    "required": true,
                                    "options": [
                                        {
                                            "id": "age_coarse",
                                            "label": "Age <15 / 15+"
                                        },
                                        {
                                            "id": "sex_age_coarse",
                                            "label": "Sex and age <15 / 15+"
                                        },
                                        {
                                            "id": "age_group",
                                            "label": "5-year age group"
                                        },
                                        {
                                            "id": "sex_age_group",
                                            "label": "Sex and 5-year age group"
                                        }
                                    ],
                                    "value": "sex_age_coarse",
                                    "helpText": "Calibrate to Spectrum mean total new infections"
                                }
                            ]
                        },
                        {
                            "label": "Calibration method",
                            "controls": [
                                {
                                    "name": "calibrate_method",
                                    "type": "select",
                                    "value": "logistic",
                                    "required": true,
                                    "options": [
                                        {
                                            "id": "logistic",
                                            "label": "Logistic"
                                        },
                                        {
                                            "id": "proportional",
                                            "label": "Proportional"
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        "id": "cdb308b0d5a0774c75e788c2547d784b"
    },
    "model_output": {
        "id": ""
    },
    "coarse_output": {
        "id": ""
    },
    "summary_report": {
        "id": ""
    },
    "comparison_report": {
        "id": ""
    },
    "version": {
        "hintr": "1.0.35",
        "naomi": "2.6.26",
        "rrq": "0.5.6",
        "traduire": "0.0.6"
    }
}

**Comment by LekanAnni** [comment]
> {
>     "datasets": {
>         "pjnz": {
>             "path": "uploads/72A9B1F58AAA743ADA64C6AE985CF228.pjnz",
>             "filename": "demo_mwi2019.pjnz"
>         },
>         "population": {
>             "path": "uploads/651105353D7153381ED29363DF0D772F.csv",
>             "filename": "demo_population_agesex.csv"
>         },
>         "shape": {
>             "path": "uploads/EBE533976BFAF0CABCA2C2E1B611B9C7.geojson",
>             "filename": "demo_areas.geojson"
>         },
>         "survey": {
>             "path": "uploads/F669CA9AA38A3993B5A9E9D3EB717C7D.csv",
>             "filename": "demo_survey_hiv_indicators.csv"
>         },
>         "programme": {
>             "path": "uploads/8301300AB39BE177FA593571B9DD94C4.csv",
>             "filename": "demo_art_number.csv"
>         },
>         "anc": {
>             "path": "uploads/E6323AAEB045D31E4A267398F669CF20.csv",
>             "filename": "demo_anc_testing.csv"
>         }
>     },
>     "model_fit": {
>         "options": {
>             "area_scope": "MWI",
>             "area_level": "4",
>             "calendar_quarter_t2": "CY2021Q4",
>             "calendar_quarter_t3": "CY2022Q3",
>             "calendar_quarter_t1": "CY2016Q1",
>             "survey_prevalence": "DEMO2016PHIA",
>             "survey_art_coverage": "DEMO2016PHIA",
>             "anc_clients_year2": "",
>             "anc_clients_year2_num_months": "12",
>             "anc_prevalence_year1": "2016",
>             "anc_prevalence_year2": "",
>             "anc_art_coverage_year1": "2016",
>             "anc_art_coverage_year2": "",
>             "include_art_t1": "true",
>             "include_art_t2": "true",
>             "artattend": "true",
>             "artattend_t2": "true",
>             "spectrum_population_calibration": "subnational",
>             "output_aware_plhiv": "true",
>             "max_iterations": 250,
>             "no_of_samples": 1000,
>             "rng_seed": 28,
>             "artattend_log_gamma_offset": -4,
>             "rho_paed_x_term": "false",
>             "rho_paed_15to49f_ratio": "true",
>             "alpha_xst_term": "false",
>             "adjust_area_growth": "false",
>             "use_survey_aggregate": "false",
>             "survey_recently_infected": [
> 
>             ],
>             "use_kish_prev": "true",
>             "deff_prev": 1,
>             "use_kish_artcov": "true",
>             "deff_artcov": 1,
>             "use_kish_recent": "true",
>             "deff_recent": 1
>         },
>         "id": "9d77425e24f7d5240543861ddf144515"
>     },
>     "calibrate": {
>         "options": {
>             "controlSections": [
>                 {
>                     "label": "Calibration options",
>                     "description": "Options for calibrating to Spectrum outputs. If national Spectrum file, 'National' and 'Subnational' options will return the same result. Some calibrations will invalidate model inference to survey, ART programme, and ANC testing data. If calibration adjustments are large, it is recommended to review Spectrum results and Naomi input data rather than accept large calibration adjustments.",
>                     "controlGroups": [
>                         {
>                             "label": "Adjust to Spectrum PLHIV",
>                             "controls": [
>                                 {
>                                     "name": "spectrum_plhiv_calibration_level",
>                                     "label": "Level",
>                                     "type": "select",
>                                     "required": true,
>                                     "options": [
>                                         {
>                                             "id": "none",
>                                             "label": "None"
>                                         },
>                                         {
>                                             "id": "national",
>                                             "label": "National"
>                                         },
>                                         {
>                                             "id": "subnational",
>                                             "label": "Subnational"
>                                         }
>                                     ],
>                                     "value": "national",
>                                     "helpText": "Calibrate to Spectrum mean total PLHIV."
>                                 },
>                                 {
>                                     "name": "spectrum_plhiv_calibration_strat",
>                                     "label": "Stratification",
>                                     "type": "select",
>                                     "required": true,
>                                     "options": [
>                                         {
>                                             "id": "age_coarse",
>                                             "label": "Age <15 / 15+"
>                                         },
>                                         {
>                                             "id": "sex_age_coarse",
>                                             "label": "Sex and age <15 / 15+"
>                                         },
>                                         {
>                                             "id": "age_group",
>                                             "label": "5-year age group"
>                                         },
>                                         {
>                                             "id": "sex_age_group",
>                                             "label": "Sex and 5-year age group"
>                                         }
>                                     ],
>                                     "value": "sex_age_coarse",
>                                     "helpText": "Calibrate to Spectrum mean total PLHIV."
>                                 }
>                             ]
>                         },
>                         {
>                             "label": "Adjust to Spectrum number on ART",
>                             "controls": [
>                                 {
>                                     "name": "spectrum_artnum_calibration_level",
>                                     "label": "Level",
>                                     "type": "select",
>                                     "required": true,
>                                     "options": [
>                                         {
>                                             "id": "none",
>                                             "label": "None"
>                                         },
>                                         {
>                                             "id": "national",
>                                             "label": "National"
>                                         },
>                                         {
>                                             "id": "subnational",
>                                             "label": "Subnational"
>                                         }
>                                     ],
>                                     "value": "national",
>                                     "helpText": "Calibrate to Spectrum mean total on ART"
>                                 },
>                                 {
>                                     "name": "spectrum_artnum_calibration_strat",
>                                     "label": "Stratification",
>                                     "type": "select",
>                                     "required": true,
>                                     "options": [
>                                         {
>                                             "id": "age_coarse",
>                                             "label": "Age <15 / 15+"
>                                         },
>                                         {
>                                             "id": "sex_age_coarse",
>                                             "label": "Sex and age <15 / 15+"
>                                         },
>                                         {
>                                             "id": "age_group",
>                                             "label": "5-year age group"
>                                         },
>                                         {
>                                             "id": "sex_age_group",
>                                             "label": "Sex and 5-year age group"
>                                         }
>                                     ],
>                                     "value": "sex_age_coarse",
>                                     "helpText": "Calibrate to Spectrum mean total on ART"
>                                 }
>                             ]
>                         },
>                         {
>                             "label": "Adjust to Spectrum number aware of HIV status",
>                             "controls": [
>                                 {
>                                     "name": "spectrum_aware_calibration_level",
>                                     "label": "Level",
>                                     "type": "select",
>                                     "required": true,
>                                     "options": [
>                                         {
>                                             "id": "none",
>                                             "label": "None"
>                                         },
>                                         {
>                                             "id": "national",
>                                             "label": "National"
>                                         },
>                                         {
>                                             "id": "subnational",
>                                             "label": "Subnational"
>                                         }
>                                     ],
>                                     "value": "national",
>                                     "helpText": "Calibrate to Spectrum mean total number aware of HIV status from Shiny90"
>                                 },
>                                 {
>                                     "name": "spectrum_aware_calibration_strat",
>                                     "label": "Stratification",
>                                     "type": "select",
>                                     "required": true,
>                                     "options": [
>                                         {
>                                             "id": "age_coarse",
>                                             "label": "Age <15 / 15+"
>                                         },
>                                         {
>                                             "id": "sex_age_coarse",
>                                             "label": "Sex and age <15 / 15+"
>                                         },
>                                         {
>                                             "id": "age_group",
>                                             "label": "5-year age group"
>                                         },
>                                         {
>                                             "id": "sex_age_group",
>                                             "label": "Sex and 5-year age group"
>                                         }
>                                     ],
>                                     "value": "sex_age_coarse",
>                                     "helpText": "Calibrate to Spectrum mean total number aware of HIV status from Shiny90"
>                                 }
>                             ]
>                         },
>                         {
>                             "label": "Adjust to Spectrum number of new infections",
>                             "controls": [
>                                 {
>                                     "name": "spectrum_infections_calibration_level",
>                                     "label": "Level",
>                                     "type": "select",
>                                     "required": true,
>                                     "options": [
>                                         {
>                                             "id": "none",
>                                             "label": "None"
>                                         },
>                                         {
>                                             "id": "national",
>                                             "label": "National"
>                                         },
>                                         {
>                                             "id": "subnational",
>                                             "label": "Subnational"
>                                         }
>                                     ],
>                                     "value": "national",
>                                     "helpText": "Calibrate to Spectrum mean total new infections"
>                                 },
>                                 {
>                                     "name": "spectrum_infections_calibration_strat",
>                                     "label": "Stratification",
>                                     "type": "select",
>                                     "required": true,
>                                     "options": [
>                                         {
>                                             "id": "age_coarse",
>                                             "label": "Age <15 / 15+"
>                                         },
>                                         {
>                                             "id": "sex_age_coarse",
>                                             "label": "Sex and age <15 / 15+"
>                                         },
>                                         {
>                                             "id": "age_group",
>                                             "label": "5-year age group"
>                                         },
>                                         {
>                                             "id": "sex_age_group",
>                                             "label": "Sex and 5-year age group"
>                                         }
>                                     ],
>                                     "value": "sex_age_coarse",
>                                     "helpText": "Calibrate to Spectrum mean total new infections"
>                                 }
>                             ]
>                         },
>                         {
>                             "label": "Calibration method",
>                             "controls": [
>                                 {
>                                     "name": "calibrate_method",
>                                     "type": "select",
>                                     "value": "logistic",
>                                     "required": true,
>                                     "options": [
>                                         {
>                                             "id": "logistic",
>                                             "label": "Logistic"
>                                         },
>                                         {
>                                             "id": "proportional",
>                                             "label": "Proportional"
>                                         }
>                                     ]
>                                 }
>                             ]
>                         }
>                     ]
>                 }
>             ]
>         },
>         "id": "0fc592026c6a1104249892a730659ba0"
>     },
>     "version": {
>         "hintr": "1.0.35",
>         "naomi": "2.6.26",
>         "rrq": "0.5.6",
>         "traduire": "0.0.6"
>     }
> }

Data is now returned from output zip as seen here

**Review by EmmaLRussell — APPROVED** [review]
LGTM, thanks for fixing the bug around setting calibrate options. Can be merged once change is available in hintr master.

**Comment by r-ash** [comment]
https://github.com/mrc-ide/hintr/pull/386 merged!

File: src/app/static/src/app/generated.d.ts [hunk 1]
```diff
@@ -385,6 +385,64 @@ export interface DownloadStatusResponse {
       }
     | string)[];
 }
+export interface DownloadSubmitRequest {
+  notes?: {
+    project_notes?: Note;
```
Comment (Nicolas-Dolan) [inline src/app/static/src/app/generated.d.ts:390]:
```suggestion
    project_note?: Note;
```
Should probably be singular as it is a single item, not an array. Here and anywhere else it shows up.

**Review by Nicolas-Dolan — APPROVED** [review]
Looks good. Just one optional change suggested.

File: src/app/static/src/app/generated.d.ts (hunk 1 above)
Comment (LekanAnni) [reply src/app/static/src/app/generated.d.ts:390]: This was generated from the schema @r-ash

**Pull request merged**